src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -139,5 +139,98 @@ public static async Task TestBOMWithNoJsonValue()
                     async () => await JsonSerializer.DeserializeAsync<byte>(stream));
             }
         }
+
+        public static IEnumerable<object[]> BOMWithJsonElementTestData
+        {
+            get
+            {
+                yield return CreateStreamArgs(100);
+                yield return CreateStreamArgs(200);
+                yield return CreateStreamArgs(400);
+                yield return CreateStreamArgs(800);
+                yield return CreateStreamArgs(1600);
+
+                static object[] CreateStreamArgs(int count)
+                {
+                    byte[] objBytes = Encoding.UTF8.GetBytes(
+                        @"{""Test"":{},""Test2"":[],""Test3"":{""Value"":{}},""PersonType"":0,""Id"":2}");
+
+                    byte[] utf8Bom = Encoding.UTF8.GetPreamble();
+                    byte[] comma = Encoding.UTF8.GetBytes(@",");
+                    byte[] startArray = Encoding.UTF8.GetBytes(@"[");
+                    byte[] endArray = Encoding.UTF8.GetBytes(@"]");
+
+                    var stream = new MemoryStream();
+
+                    stream.Write(utf8Bom, 0, utf8Bom.Length);
+                    stream.Write(startArray, 0, startArray.Length);
+
+                    for (int i = 1; i <= count; i++)
+                    {
+                        stream.Write(objBytes, 0, objBytes.Length);
+
+                        if (i < count)
+                        {
+                            stream.Write(comma, 0, comma.Length);
+                        }
+                    }
+
+                    stream.Write(endArray, 0, endArray.Length);
+
+                    stream.Position = 0;
+                    return new object[] { stream, count };
+                }
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(BOMWithJsonElementTestData))]
+        public static async Task TestBOMWithShortAndLongBuffers(Stream stream, int count)
```

Comment:
Are there any cases, outside of when the payload contains a BOM that would be also worthwhile testing here (so that we hit the `bytesInBuffer != 0` code path)?

Reply:
That code path is already tested elsewhere; just was not tested with the BOM (`start != 0`)